pkg/hardware/yamlparser.go

```diff
@@ -0,0 +1,146 @@
+package hardware
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	pbnjv1alpha1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/pbnj/api/v1alpha1"
+	tinkv1alpha1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/tink/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	yamlPath      = "hardware-manifests/hardware.yaml"
+	yamlSeparator = "---\n"
+)
+
+type YamlParser struct {
+	file *os.File
+}
+
+func NewYamlParser() (*YamlParser, error) {
+	err := ioutil.WriteFile(yamlPath, []byte{}, 0o644)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing YamlParser: %v", err)
+	}
+
+	file, err := os.OpenFile(yamlPath, os.O_APPEND|os.O_WRONLY, 0o644)
+	if err != nil {
+		file.Close()
+		return nil, fmt.Errorf("error initializing YamlParser: %v", err)
+	}
+	return &YamlParser{
+		file: file,
+	}, nil
+}
+
+func (y *YamlParser) Close() {
+	y.file.Close()
+}
+
+func (y *YamlParser) WriteHardwareYaml(id, hostname, bmcIp, vendor, username, password string) error {
+	bmcRef := fmt.Sprintf("bmc-%s", hostname)
+	secretRef := fmt.Sprintf("%s-auth", bmcRef)
+	hardware := tinkv1alpha1.Hardware{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Hardware",
+			APIVersion: "tinkerbell.org/v1alpha1",
```

Comment:
Not sure, but maybe we can use these GroupVersions defined here?
[groupversion_info.go](https://github.com/pokearu/cluster-api-provider-tinkerbell/blob/release-v0.1.0/pbnj/api/v1alpha1/groupversion_info.go#L30)

Reply:
Any strong opinion on this? I would prefer to keep it simple and not introduce additional dependencies just for this API version. We also keep this static in all the other configs like [this one](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/config/template-cp.yaml#L1)